src/ferui-components/tree-view/tree-view-component.ts

```diff
@@ -0,0 +1,399 @@
+        left: 50%;
+        color: #87a1b5;
+        text-align: center;
+        transform: translateX(-50%);
+      }
+      .DARK_BLUE {
+        background: #252a3a;
+        color: #fff;
+      }
+      .LIGHT_BLUE {
+        background: #03a6ff;
+        color: #fff;
+      }
+      .GRAY {
+        background: #f5f8f9;
+        color: #252a3a;
+      }
+      .WHITE {
+        background: #fff;
+        color: #252a3a;
+      }
+      .fui-virtual-scroller {
+        height: 100%;
+        width: 100%;
+      }
+      :host(.show-border) #fui-tree-view-component {
+        border: #e5eaee 1px solid;
+        padding: 0;
+      }
+    `,
+  ],
+})
+export class FuiTreeViewComponent<T> implements OnInit, OnDestroy {
+  @Input() treeNodeData: TreeNodeData<T>;
+
+  @Input() dataRetriever: TreeNodeDataRetriever<T> | PagedTreeNodeDataRetriever<T>;
+
+  @Input() config: TreeViewConfiguration;
+
+  @Input() loading: boolean = false;
+
+  @Input() error: boolean = false;
+
+  @HostBinding('class.show-border') border: boolean;
+
+  @ViewChild(FuiVirtualScrollerComponent) vs: FuiVirtualScrollerComponent;
+
+  @ViewChildren(FuiTreeNodeComponent) children: QueryList<FuiTreeNodeComponent<T>>;
+
+  @Output() onNodeEvent: EventEmitter<TreeViewEvent<any>> = new EventEmitter<TreeViewEvent<any>>();
+
+  treeViewStyles: FuiTreeViewComponentStyles;
+  colorTheme: TreeViewColorTheme;
+  scrollViewArray: TreeNode<T>[];
+  hasBorders: boolean;
+  BUFFER_AMOUNT: number = 1000;
+  private rootNode: TreeNode<T>;
+  private nonRootArray: TreeNode<T>[];
+  private scrollObservable: Observable<any>;
+  private scrollSubscription: Subscription;
+  private indexChangeSubscription: Subscription;
+  private originalWidth: number;
+  private scrollPromise: boolean = false;
+  private SERVER_SIDE_COMPONENT: boolean = false;
+  private TREE_NODE_COMPONENT_HEIGHT: number = 34;
+  private DEFAULT_WIDTH: string = 'auto';
+  private DEFAULT_HEIGHT: string = '100%';
+  private DEFAULT_COLOR_THEME: TreeViewColorTheme = 'WHITE';
+  // Default Throttle time fro scroll event
+  private THROTTLE_TIME: number = 500;
+
+  constructor(private el: ElementRef) {}
+
+  /**
+   * Initializes the Tree View component and all properties needed depending on inputs configuration
+   */
+  ngOnInit(): void {
+    this.SERVER_SIDE_COMPONENT = this.dataRetriever.hasOwnProperty('getPagedChildNodeData');
+    this.treeViewStyles = {
+      width: this.config.width ? this.config.width : this.DEFAULT_WIDTH,
+      height: this.config.height ? this.config.height : this.DEFAULT_HEIGHT,
+    };
+    this.indexChangeSubscription = this.vs.vsChange.subscribe(() => this.calculateScrollerWidth());
+    this.colorTheme = this.config.colorVariation ? this.config.colorVariation : this.DEFAULT_COLOR_THEME;
+    this.border = this.hasBorders = this.config.hasBorders ? this.config.hasBorders : false;
+    if (this.SERVER_SIDE_COMPONENT) {
+      this.BUFFER_AMOUNT = this.config.bufferAmount || this.BUFFER_AMOUNT;
```

Comment:
Please, do not add an extra observable for scroll event. You can use the existing `this.vs.vsChange.subscribe(pageInfo => {});` subscription to track whenever the user scroll within the virtual-scroller like we've talked about last time.

Reply:
Jon the vsChange observable happens when we expand and collapse the nodes as well so it is difficult to differentiate what event is happening since this observable only returns the page info

Reply:
@JoMas971 I updated this to use the vsChange and filtering when the view is full to know the user is scrolling